### PR TITLE
Add definitions for react-modal-v3.12.x

### PR DESCRIPTION
**Pull request by knu**
- Links to documentation: https://github.com/reactjs/react-modal/blob/master/CHANGELOG.md#3121---sun-22-nov-2020-225112-utc
- Link to GitHub or NPM: https://www.npmjs.com/package/react-modal
- Type of contribution: addition

### CODE REVIEW

File: definitions/npm/react-modal_v3.12.x/flow_v0.104.x-/react-modal_v3.12.x.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
+declare module 'react-modal' {
+  declare type ModalElementProps = {
+    ref: (node: ?HTMLElement) => mixed,
+    id: string,
+    className: string,
+    style: { [key: string]: string | number },
+    onClick: (event: SyntheticMouseEvent<*>) => void,
+    onMouseDown: (event: SyntheticMouseEvent<*>) => void,
+  };
```
Comment (pascalduez) [inline definitions/npm/react-modal_v3.12.x/flow_v0.104.x-/react-modal_v3.12.x.js:9]: You'd need to mark that object as either exact or inexact.

File: definitions/npm/react-modal_v3.12.x/flow_v0.104.x-/react-modal_v3.12.x.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
+declare module 'react-modal' {
+  declare type ModalElementProps = {
+    ref: (node: ?HTMLElement) => mixed,
+    id: string,
+    className: string,
+    style: { [key: string]: string | number },
```
Comment (pascalduez) [inline definitions/npm/react-modal_v3.12.x/flow_v0.104.x-/react-modal_v3.12.x.js:6]: You'd need to mark that object as either exact or inexact.

File: definitions/npm/react-modal_v3.12.x/flow_v0.104.x-/react-modal_v3.12.x.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    onClick: (event: SyntheticMouseEvent<*>) => void,
```
Comment (pascalduez) [inline definitions/npm/react-modal_v3.12.x/flow_v0.104.x-/react-modal_v3.12.x.js:7]: The existential type `*` is deprecated, probably `any` is enough here.

Comment (pascalduez) [reply definitions/npm/react-modal_v3.12.x/flow_v0.104.x-/react-modal_v3.12.x.js:7]: Event callback returns types are usually better typed as `mixed`, accept async callbacks.

File: definitions/npm/react-modal_v3.12.x/test_react-modal_v3.12.x.js [hunk 4]
```diff
@@ -0,0 +1,163 @@
+// @flow
+
+import React from 'react';
```
Comment (pascalduez) [inline definitions/npm/react-modal_v3.12.x/test_react-modal_v3.12.x.js:3]:
```suggestion
import * as React from 'react';
```

**Comment by knu** [comment]
Thanks for the review comments!  I added a commit to address them.  That was helpful as existing type definitions do not say much about what the latest practices are.

**Review by pascalduez — APPROVED** [review]
Thanks!

**Pull request merged**